packages/back-end/src/jobs/updateScheduledFeatures.ts

```diff
@@ -0,0 +1,102 @@
+import Agenda, { Job } from "agenda";
+import { FeatureModel } from "../models/FeatureModel";
+import {
+  featureUpdated,
+  getEnabledEnvironments,
+  getNextScheduledUpdate,
+} from "../services/features";
+import { logger } from "../util/logger";
+
+type UpdateSingleFeatureJob = Job<{
+  featureId: string;
+}>;
+
+const QUEUE_FEATURE_UPDATES = "queueScheduledFeatureUpdates";
+
+const UPDATE_SINGLE_FEATURE = "updateSingleFeature";
+
+async function fireUpdateWebhook(agenda: Agenda) {
+  const updateFeatureJob = agenda.create(QUEUE_FEATURE_UPDATES, {});
+  updateFeatureJob.unique({});
+  updateFeatureJob.repeatEvery("1 minute");
+  await updateFeatureJob.save();
+}
+
+async function queueFeatureUpdate(agenda: Agenda, featureId: string) {
+  const job = agenda.create(UPDATE_SINGLE_FEATURE, {
+    featureId,
+  }) as UpdateSingleFeatureJob;
+
+  job.unique({
+    featureId,
+  });
+  job.schedule(new Date());
+  await job.save();
+}
+
+export default async function (agenda: Agenda) {
+  agenda.define(QUEUE_FEATURE_UPDATES, async () => {
+    const featureIds = (
+      await FeatureModel.find({
+        nextScheduledUpdate: {
+          $exists: true,
+          $lt: new Date(),
+        },
+      })
+    ).map((f) => f.id);
+
+    for (let i = 0; i < featureIds.length; i++) {
+      await queueFeatureUpdate(agenda, featureIds[i]);
+    }
+  });
+
+  agenda.define(
+    "updateSingleFeature",
+    { lockLifetime: 30 * 60 * 1000 },
+    updateSingleFeature
+  );
+
+  await fireUpdateWebhook(agenda);
+}
+
+async function updateSingleFeature(job: UpdateSingleFeatureJob) {
+  // Get the feature from the DB
+  const featureId = job.attrs.data?.featureId;
+  if (!featureId) return;
+
+  const log = logger.child({
+    cron: "updateSingleFeature",
+    featureId,
+  });
+
+  const feature = await FeatureModel.findOne({
```

Comment:
We always need an organization id when looking up features.  Ids alone are not unique